### ISSUE TITLE
(Non) Screen reader bug

### ISSUE THREAD

**Issue by andreantunes2602** [glpi-project/glpi#20828]
### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

10.0.18

### Bug description

Greetings community,

Over the last two weeks, we have been experiencing a situation with an employee who is having some difficulties with the accessibility features in GLPI 10.0.18.

This employee has a visual impairment and uses a screen reader (NVDA) to work.

Most features work well, but he is having issues when the screen reader cannot recognize the category menu once it is selected.

Instead, the screen reader reports it as a blank field.

We suspect this may be caused by the field’s behavior when clicked, since the ability to search a category in the menu seems to make it appear blank.

We are working with all of our knowledge to try to help him, but we would really appreciate suggestions from the community.

Does the community have any ideas on how to make the screen reader properly read this menu?

Perhaps this could also be a good improvement to consider for future releases, to help others as well.

We count on the community’s collaboration to help this person.

Best regards,

Andre Antunes 

### Relevant log output

```shell

```

### Steps To reproduce

Using NVDA open a new ticket 
Check if when choosing categories NVDA says something like "blank field" or something like this.

**Comment by cconard96** [comment]
I'm not confident there is much that can be done here easily from the GLPI side, although someone else may have some idea for a fix. The library used for the dropdowns in GLPI, `select2` is known to have accessibility issues.

Issues related to "blank field":
https://github.com/select2/select2/issues?q=is%3Aissue%20state%3Aopen%20blank%20field

All tagged accessibility issues:
https://github.com/select2/select2/issues?q=is%3Aissue%20state%3Aopen%20label%3Aaccessibility

**Comment by trasher** [comment]
I just made a quick try with latest GLPI v11 nightly, and I guess I did not reproduce (I've never used a screen-reader before).

GLPI v11 is a major rewrite, many changes has been made UI side. Maybe could you test with v11 if that issue is solved?

**Comment by andreantunes2602** [comment]
Thanks man!!! I'll read it ASAP.

**Comment by andreantunes2602** [comment]
Thank you @trasher 
I'll test this version

**Comment by andreantunes2602** [comment]
Hi community.

I've tested the last version 11(rc3) and no changes has been seen on category select field.

Shoud this post can be change to a feature request? If this change is possible, i have this sugestions:

-- Search field be filled by the selected option onmouse hover
OR
-- User option for disable search in select field.

Any of this could be appreciated by our users with visual impairment.

Thanks for your attention and have a nice day.

**Comment by trasher** [comment]
> I've tested the last version 11(rc3) and no changes has been seen on category select field.

:(

> -- Search field be filled by the selected option onmouse hover 

I guess we may try that, we'll keep you in touch.

> OR -- User option for disable search in select field.

Some dropdowns can be very huge, that does not seems a good idea.

**Comment by andreantunes2602** [comment]
Thank you all
It should be a good workaround for screen readers working well.

Have a nice day

**Comment by citygabor** [comment]
Hello GLPI Community and GLPI Developers,

I could confirm the same the new version of GLPI 11.x version has the same  accessibility issues as it has before. I think nothing changed.
I am going to open a separated issue with my listed issues and suggestions.

That would be really good to have aria labels and title for the form and buttons elements as this is the key aspect for all of the well-known  screen readers like: JAWS, NVDA, Narrator, Voice over, etc..

I was trying to find the HTML buttons and form elements in your repository, but it is really hard to find it if I am not really familier with programing in PHP or not familiar with the structure of the files and the codes.

I could really suggest to use the Accessibility Insight  for Web Extension for validate the basic screen reader issues or accessibility issues before the new version is out.

https://accessibilityinsights.io/docs/web/overview/


Please let me know if you need anything from my side  I am also a screen reader user as that person colleague of the reporter.

Thanks for your support,
regards,
gabor

Please let me know

**Comment by andreantunes2602** [comment]
Hi there.

Does we have some new feature about creen readers?
I made a test on v11 but had no success.

It makes me happy if we have some answer.

Thanks for all.

**Comment by trasher** [comment]
Hi,

We began to work a little on accessibility; v11 will unfortunately never be compliant.
Some more work has been done on the main branch; but that's far to be done. 3rd party libraries are another problem (there is a PR currently in review to update select2 that has make some changes on their side, I have no real idea if it will be OK or not :/).
